task_10/src/components/LocationSearch.js

```diff
@@ -0,0 +1,58 @@
+import {getForecastFromApi} from "../utils/api";
+import {setCoordinatesToMapStorage} from "../utils/setCoordinates";
+import {HandlingURL} from "../utils/Url";
+import {currentUserPosition} from "../utils/settings";
+import {RecentlyCities} from "./RecentlyCities";
+
+class LocationSearch {
+    constructor(props) {
+        this.state = {
+            isValid: true
+        };
+        this.props = props;
+        this.handleSubmit = this.handleSubmit.bind(this);
+        this.host = document.getElementById('input-search-container');
+        this.host.addEventListener('submit', this.handleSubmit);
+    }
+
+    updateState(nextState) {
+        this.state = nextState;
+    }
+
+    handleSubmit(ev) {
+        ev.preventDefault();
+        const city = ev.target.elements.search.value.trim();
+        if (!city.length) {
+            this.updateState({isValid: false});
+        } else {
+            const geocoder = new google.maps.Geocoder();
```

Comment:
You can create instance of geocoder only once and then use it.

Also, can you move this logic to separate function? This will make code cleaner and more self-descriptive.